analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -749,6 +749,93 @@ public static class Guard
+                        Dim SomeString As String = "Lorem"
+                        Guard.NotNullExt(o1)
+                        o2.NotNullExt()
+                        NotNullInst(o3)
+                        NotNullInst(o4, o5)
+                        NotNullInst(value2:=o6, value1:=o7, value3:=o8)  ' value2 is not annotated
+                        NotNullRef(s1)
+                        SomeString.NotNullExtOtherArgument(s2)
+                        ex.NotNullExt()
+                        Tag("AfterGuard_o1", o1)
+                        Tag("AfterGuard_o2", o2)
+                        Tag("AfterGuard_o3", o3)
+                        Tag("AfterGuard_o4", o4)
+                        Tag("AfterGuard_o5", o5)
+                        Tag("AfterGuard_o6", o6)
+                        Tag("AfterGuard_o7", o7)
+                        Tag("AfterGuard_o8", o8)
+                        Tag("AfterGuard_s1", s1)
+                        Tag("AfterGuard_s2", s2)
+                        Tag("AfterGuard_ex", ex)
+                    End Sub
+
+                    Private Sub NotNullInst({notNullAttribute} value As Object)
+                        ' Skip implementation to make sure, the attribute Is driving the constraint
+                    End Sub
+
+                    Private Sub NotNullInst({notNullAttribute} value1 As Object, {notNullAttribute} value2 As Object)
+                    End Sub
+
+                    Private Sub NotNullInst(Of T1, T2, T3)({notNullAttribute} value1 As T1, value2 As T2, {notNullAttribute} value3 As T3)
+                    End Sub
+
+                    Private Sub NotNullRef(Of T)({notNullAttribute} ByRef value As T)
+                    End Sub
+
+                    Private Shared Sub Tag(name As String, arg As Object)
+                    End Sub
+
+                End Class
+
+                Public NotInheritable Class ValidatedNotNullAttribute
+                    Inherits Attribute
+                End Class
+
+                Public NotInheritable Class NotNullAttribute
+                    Inherits Attribute
+                End Class
+
+                Public Module Guard
+
+                    <Extension>
+                    Public Sub NotNullExt(Of T As Class)({notNullAttribute} Value As T)
+                    End Sub
+
+                    <Extension>
+                    Public Sub NotNullExtOtherArgument(Of T As Class)(Value As String, {notNullAttribute} NotNull As T)
+                    End Sub
+
+                End Module
+                """;
+            var validator = new SETestContext(code, AnalyzerLanguage.VisualBasic, Array.Empty<SymbolicCheck>()).Validator;
+            validator.ValidateTag("AfterGuard_o1", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+            validator.ValidateTag("AfterGuard_o2", x => x.Should().HaveNoConstraints());    // Extension method invoked on Object type is DynamicInvocationOperation that we don't support
```

Comment:
The comment could also go into "because" as there isn't much we can do about it and it will be permanent behavior.
```suggestion
            validator.ValidateTag("AfterGuard_o2", x => x.Should().HaveNoConstraints("extension method invoked on Object type is DynamicInvocationOperation that we don't support"));
```